scipy/stats/morestats.py

```diff
@@ -2078,18 +2078,86 @@ def anderson_ksamp(samples, midrank=True):
 AnsariResult = namedtuple('AnsariResult', ('statistic', 'pvalue'))
 
 
-def ansari(x, y):
+class _ABW:
```

Comment:
Next time I should use `rv_discrete` to create the distribution, if possible! It didn't occur to me before.

On the other hand, part of this is to be careful with the rounding behavior for non-integer inputs, so depending on how `rv_discrete` does that, this might not have saved much code.

Reply:
> Next time I should use `rv_discrete` to create the distribution, if possible!

Agreed!

> On the other hand, part of this is to be careful with the rounding behavior for non-integer inputs, so depending on how `rv_discrete` does that by default, it might not have saved much code.

I also think that `rv_discrete` is not of much help here and we probably won't need the extra methods it implements.